### PR TITLE
make test individually runable

### PR DESCRIPTION
**Pull request by davidcheung**
### Description

previous when you do command such as `mocha test/model.test.js`
we get errors such as "expect is not a function", "assert is not a function"
this is to organize the test file in a slightly different way, so we can do `mocha test/model.test.js`

not sure if we need to modify [`karma.conf`](https://github.com/strongloop/loopback/blob/c28a1568b31b8fb8d75273b4940b1aa8cabc2828/test/karma.conf.js#L23) too, npm test is locally passing for me
#### Related issues
- None
### Checklist
- [ ] New tests are added to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

**Update 14/11/16**:
  - Needs re-review from @bajtos or @strongloop/squad-epitome 

**Update 14/11/16**:
  - @pthieu will finish review.

### CODE REVIEW

**Comment by davidcheung** [comment]
cc @superkhau

**Comment by davidcheung** [comment]
@bajtos do you know if there is any negative implication with this change, i have manually tried to run the tests individually and it seems to work fine. 

cc @pthieu @bosco-li

File: test/access-token.test.js [hunk 1]
```diff
@@ -15,6 +15,10 @@ var ds = loopback.createDataSource({ connector: loopback.Memory });
 Token.attachTo(ds);
 var ACL = loopback.ACL;
 
+beforeEach(function() {
+  this.app = app = loopback();
```
Comment (pthieu) [inline test/access-token.test.js]: How does this pass `eslint`? `app` would be a global here.

File: test/access-token.test.js [hunk 2]
```diff
@@ -15,6 +15,10 @@ var ds = loopback.createDataSource({ connector: loopback.Memory });
 Token.attachTo(ds);
```
Comment (pthieu) [inline test/access-token.test.js]: No `use strict`; at the top.

File: test/access-token.test.js (hunk 1 above)
Comment (superkhau) [inline test/access-token.test.js]: LoopBack is using eslint-config-loopback@^1.0.0, the use strict check was added later on -- we need to update this project to use ^4.0.0

Comment (davidcheung) [inline test/access-token.test.js]: the mocha-opts requires `support.js` which app ends up global in our test suite 
https://github.com/strongloop/loopback/blob/d85ff32abb681a127659e4ed6ab4cc0fa3fac922/test/support.js#L10-L18

Comment (bajtos) [inline test/access-token.test.js]: Please define a describe-local `var app` and move the hook into the top-level describe block, so that we don't pollute global namespace and don't run this hook for tests that don't need it.

```js
describe('loopback.token(options)', function() {
  var app;
  beforeEach(function() {
    this.app = app = loopback();
  });

  // ...
});
```

File: test/mocha.opts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+--require ./test/support.js
+--require ./test/helpers/loopback-testing-helper.js
```
Comment (bajtos) [inline test/mocha.opts]: Why is this second line required? AFAICT, all modules using `loopback-testing-helper` are directly calling `require()` to get it.

File: test/support.js [hunk 4]
```diff
@@ -21,10 +21,6 @@ var RemoteObjects = require('strong-remoting');
 // for tests using the built-in User model
 loopback.User.settings.saltWorkFactor = 4;
 
-beforeEach(function() {
-  this.app = app = loopback();
```
Comment (bajtos) [inline test/support.js]: Also remove the global `app` variable defined at line 15

File: test/access-token.test.js (hunk 2 above)
Comment (bajtos) [inline test/access-token.test.js]: I agree with add `use strict` to all files in a standalone commit.

File: test/app.test.js [hunk 5]
```diff
@@ -15,6 +15,10 @@ var describe = require('./util/describe');
 var expect = require('chai').expect;
 var it = require('./util/it');
 
+beforeEach(function() {
+  this.app = app = loopback();
```
Comment (bajtos) [inline test/app.test.js]: Same comment as above, move this code to `describe` block and define `app` as a describe-local variable.

File: test/access-token.test.js [hunk 6]
```diff
@@ -10,13 +10,17 @@ var loopback = require('../');
 var extend = require('util')._extend;
 var session = require('express-session');
 
+var app;
```
Comment (bajtos) [inline test/access-token.test.js]: Nitpick: we usually put describe-scoped variables inside a `describe` block.

File: test/model.test.js [hunk 7]
```diff
@@ -10,6 +10,7 @@ var defineModelTestsWithDataSource = require('./util/model-tests');
 var PersistedModel = loopback.PersistedModel;
 
 var describe = require('./util/describe');
+require('./helpers/loopback-testing-helper');
```
Comment (bajtos) [inline test/model.test.js]: As I commented before, I don't understand why is this needed, could you please explain?

Comment (davidcheung) [inline test/model.test.js]: i will change it to directly use those lines

**Comment by davidcheung** [comment]
@bajtos I have addressed the feedback, I think this is pretty close anyways, will ask @phong to PTAL

File: test/model.test.js [hunk 8]
```diff
@@ -10,6 +10,9 @@ var defineModelTestsWithDataSource = require('./util/model-tests');
 var PersistedModel = loopback.PersistedModel;
 
 var describe = require('./util/describe');
+var chai = require('chai');
```
Comment (pthieu) [inline test/model.test.js]: alphabetical

**Comment by pthieu** [comment]
LGTM from me after alphabetical change -- @bajtos can you approve yours if you're ok with changes.

**Comment by pthieu** [comment]
@bajtos  is on vacation for a week, @davidcheung, since @superkhau has approved, I think you can merge.

File: test/access-token.test.js (hunk 6 above)
Comment (davidcheung) [inline test/access-token.test.js]: addressed
